src/control/system/membership.go

```diff
@@ -711,6 +711,11 @@ func (m *Membership) handleEngineFailure(evt *events.RASEvent) {
 		return
 	}
 
+	if evt.Incarnation != 0 && evt.Incarnation < member.Incarnation {
```

Comment:
It's good that we use `<` instead of `==` here, since there's subtlety with our incarnation numbers today:
```
Engine X joins with incarnation 1
Engine X bumps its incarnation from 1 to 2 because it finds that someone suspects it (The control plane is not informed directly about this.)
Engine X terminates
MS sees events for engine X with incarnation 2
```